readme.md

```diff
@@ -0,0 +1,176 @@
+# Tech Spec - Teleport Challenge
+
+## Overview
+To be implemented:
+
+1. `Job` library for starting, stopping, and getting status and streamed logs of jobs.
+2. API that implements the `Job` library, permitting clients to call the 4 public methods.
+3. CLI Client that calls API, using command line args to specify API methods.
+
+## Job Library
+`Job` type will include public fields:
+
+- Command
+- Args
+- ProcessID
+
+##### Implement methods for:
+
+- Start
+	- Assure job command exists. Executables can be located in Go using `exec.LookPath()`.
+	- Assure that a duplicate command is not currently running by checking "DB".
+	- Execute job w/ args.
+	- Write job stdout & stderr to specified dir.
+	- Set `ProcessID` on `Job`.
+- Stop
+	- Kill process.
+	- Delete job from "DB".
+- Get status
+	- Check "DB".  
+	- Return "running/complete" and logs.
+- Stream logs
+
+##### Tradeoffs
+- Killing the process, rather than sending SIGINT/SIGQUIT and waiting, comes with the risks associated with kill, e.g. lingering resources. However, it seems like the most user-intuitive option. Future versions could permit users to specify how they want to "stop" a job.
+- This implemenatation will prevent duplicate jobs from running simultaneously. There is conceivably some use in permitting redundant jobs, but it requires a more discussion around use cases (e.g. will this tool be used by customers in automation?)
```

Comment:
How would redundant jobs be detected and why is that useful? If I call the API endpoint two times with the same command, I'd expect two jobs to be started.